src/client/Microsoft.Identity.Client/Internal/RequestContext.cs

```diff
@@ -26,12 +27,14 @@ internal class RequestContext
 
         public CancellationToken UserCancellationToken { get; }
 
+        public X509Certificate2 MtlsCertificate { get; set; }
```

Comment:
Can you at least inject this via the ctor? to keep the object immutable? (i.e. get rid of the setter)